src/unity/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py

```diff
@@ -656,21 +657,137 @@ def evaluate(self, dataset, metric = 'auto', verbose = True):
+            Choose the return type of the prediction:
+            - `probability`: Probability associated with each label in the 
+                             prediction.
+            - `rank`       : Rank associated with each label in the prediction.
+            
+        k : int, optional
+            Number of classes to return for each input example.
+
+        batch_size : int, optional
+            If you are getting memory errors, try decreasing this value. If you
+            have a powerful computer, increasing this value may improve
+            performance.
+
+        Returns
+        -------
+        out : SFrame
+            An SFrame with model predictions.
+
+        See Also
+        --------
+        predict, evaluate
+
+        Examples
+        --------
+        >>> pred = m.predict_topk(validation_data, k=3)
+        >>> pred
+        +----+-------+-------------------+
+        | id | class |   probability     |
+        +----+-------+-------------------+
+        | 0  |   4   |   0.995623886585  |
+        | 0  |   9   |  0.0038311756216  |
+        | 0  |   7   | 0.000301006948575 |
+        | 1  |   1   |   0.928708016872  |
+        | 1  |   3   |  0.0440889261663  |
+        | 1  |   2   |  0.0176190119237  |
+        | 2  |   3   |   0.996967732906  |
+        | 2  |   2   |  0.00151345680933 |
+        | 2  |   7   | 0.000637513934635 |
+        | 3  |   1   |   0.998070061207  |
+        | .. |  ...  |        ...        |
+        +----+-------+-------------------+
+        [35688 rows x 3 columns]
+        """
+        _tkutl._check_categorical_option_type("output_type", output_type, 
+            ["probability", "rank"])
+        int_type_error_string = (lambda var_name: ("'{var_name}' must be " 
+            + "an integer >= 1").format(var_name=var_name))
+        non_neg_range_error_string = (lambda var_name: ("'{var_name}' must be " 
+            + ">= 1").format(var_name=var_name))
+        if not isinstance(k, int): raise TypeError(int_type_error_string("k"))
+        if k <= 0: raise ValueError(non_neg_range_error_string("k"))
+        if batch_size is not None and not isinstance(batch_size, int):
+            raise TypeError(int_type_error_string("batch_size"))
+        if batch_size is not None and batch_size < 1:
+            raise ValueError(non_neg_range_error_string("batch_size"))
+
+        prob_vector = self.predict(
+            dataset, output_type='probability_vector', batch_size=batch_size)
+
+        classes = self.classes
+        if output_type == 'probability':
+            results = prob_vector.apply(lambda p: [
+                            {'class': classes[i], 'probability': p[i]}
+                            for i in reversed(_np.argsort(p)[-k:])]
+                        )
+        else:
+            assert(output_type == 'rank')
+            def convert_probability_vector_to_ranks(p):
+                rank, ranks_list = 0, []
+                for index in reversed(_np.argsort(p)[-k:]):
```

Comment:
Could you just do `for rank, index in enumerate(reversed(_np.argsort(p)[-k:])):` and then delete line 750?

Reply:
Done.